example/lib/named_routes.g.dart

```diff
@@ -0,0 +1,58 @@
+part of 'named_routes.dart';
+
+GoRoute get homeGoRoute => GoRoute(
+      name: 'home',
+      path: '/',
+      pageBuilder: $materialPageFactory(HomeRouteExtension._fromState),
+      routes: [
+        GoRoute(
+          name: 'family',
+          path: 'family/:fid',
+          pageBuilder: $materialPageFactory(FamilyRouteExtension._fromState),
+          routes: [
+            GoRoute(
+              name: 'person',
+              path: 'person/:pid',
+              pageBuilder:
+                  $materialPageFactory(PersonRouteExtension._fromState),
+            )
+          ],
+        ),
+      ],
+    );
+
+extension HomeRouteExtension on HomeRoute {
+  static HomeRoute _fromState(GoRouterState state) => HomeRoute();
+
+  String get location => '/';
+
+  void go(BuildContext buildContext, {Object? extra}) {
+    buildContext.go(location, extra: extra);
+  }
+}
+
+extension FamilyRouteExtension on FamilyRoute {
+  static FamilyRoute _fromState(GoRouterState state) => FamilyRoute(
+        state.params['fid']!,
+      );
+
+  String get location => '/family/${Uri.encodeComponent(fid)}';
```

Comment:
@csells – so instead of having a new class `$PathData` – we'd just encode the location like this!

Then we don't need the whole `locationName` from State or Context.

This'll work, right?